docs/plugins.md

```diff
@@ -0,0 +1,79 @@
+# Train Plugins
+
+## Introducing Plugins
+
+Train plugins are a way to add new transports and platform detection to Train.
+
+If you are familiar with InSpec plugins, be forewarned; the two plugin systems are not similar.
+
+### Why Plugins?
+
+#### Benefits of plugins
+
+Plugins address two main needs that the Chef InSpec Engineering team encountered in 2017-2018.
+
+* Passionate contributors parties can develop and release new Train transports at their own pace, without gating by Chef engineers.
+* Reduction of dependency bloat within Train.  For example, since only the AWS transport need the aws-sdk, we can move the gem dependency into the plugin gem, and out of train itself.
+
+#### Future of existing Transports
+
+The Chef InSpec Engineering team currently (October 2018) plans to migrate most existing Train transports into plugins.  For example, AWS, Azure, and GCP are all excellent candidates for migration to plugin status.  The team commits to keeping SSH, WinRM, and Local transports in Train core.  All other transports may be migrated.
+
+In the near-term, InSpec will carry a gemspec dependency on the migrated plugins.  This will continue a smooth experience for users relying on (for example) Azure.
+
+## Managing Plugins
+
+### Installing and Managing Train Plugins as an InSpec User
+
+InSpec has a command-line plugin management interface, which is used for managing both InSpec plugins and Train plugins.  For example, to install `train-aws`, simply run:
+
+```bash
+$ inspec plugin install train-aws
+```
+
+The management facility can install, update, and remove plugins, including their dependencies.
+
+### Installing Train Plugins outside of InSpec
+
+If you need a train plugin installed, and `inspec plugin` is not available to you, you can install a train plugin like any other gem.  Just be sure to use the `gem` binary that comes with the application you wish to extend.  For example, to add a Train Plugin to a ChefDK installation, use:
```

Comment:
I'm not sure we support adding gems to `ChefDK`. It is intended to be a bundled distribution.

Reply:
@jquick?

Reply:
ChefDK should always have `train` but the example is fine imo.

Reply:
I only worry that a user will add a gem and pull in a dep that will break ChefDK in some way.

Reply:
If deps have issues it should not install and error just like a bundle but retain functionality. If they want to install a new Train on their DevKit that is their call.